Minor changes to tincr checkpoints for FPL

ttown523 opened the pull request:
## In this commit:
- Updated RapidSmith checkpoint formats
- Fixed a bug with a "get_part" function call that is no longer supported by Vivado
- Added a -verbose option when importing a TCP checkpoint, and added a function to print verbosely.
- Replaced several places that had tabs, with spaces
- modified the genCellLibrary.tcl to only print out leaf cell information (i.e. skip macros for now)

ttown523 committed:
Minor changes to tincr checkpoints for FPL

bradselw commented on tincr/io/design/tincr_checkpoints.tcl:
```diff
@@ -520,7 +521,7 @@ proc ::tincr::write_routing_rs2 {args} {
 				# assuming that the second tile in the tile list is the interconnect tile
 				set switchbox_tile [lindex $tiles 1]
-				set route_string [string range [get_property ROUTE $net] 3 end-3]
+				set route_string [string range [get_property ROUTE $net] 3 end-4]
```
Add a comment explaining what this line of code is doing (perhaps even include an example)

bradselw commented on tincr/io/design/tincr_checkpoints.tcl:
```diff
@@ -223,14 +223,15 @@ proc ::tincr::read_tcp {args} {
+	if {!$quiet} {
+		puts "Design importation complete. ($total_runtime seconds)"
+	}
+	# remove_files -quiet *
```
Why are you commenting out the remove_files command?

ttown523 replied:
I added the -no_add option with the read_xdc commands. I uncommented this so all files are now removed. Keep in mind, that this file is still a work in progress, and so there are several places that I have commented out for testing different things, but don't want to delete.

ttown523 committed:
refactored tincr_checkpoints, added comments

ttown523 commented:
In the latest commit, I refactored the "read_tcp" function a bit to clean some things up. Also, I added a few comments to help improve readability. There are still few changes in this pull request.

bradselw commented on tincr/cad/util/util.tcl:
```diff
@@ -51,7 +51,9 @@ namespace eval ::tincr {
         run_in_temporary_project \
-        organize_by
+        organize_by \
+		print_quiet \
```
Spaces instead of tabs

bradselw commented on tincr/cad/util/util.tcl:
```diff
@@ -1337,6 +1339,18 @@ proc ::tincr::catch_info { cmd } {
+	}
+}
+
+proc ::tincr::time_command { command } {
```
Tincr's `report_runtime` command already returns the runtime of any given command (in various formats, including seconds). Remove `time_command` and use `report_runtime <command> s` instead.

bradselw commented on tincr/cad/util/util.tcl:
```diff
@@ -1337,6 +1339,18 @@ proc ::tincr::catch_info { cmd } {
 #  namespace import ::tincr::*
 interp alias {} ::tincr::import_all {} namespace import ::tincr::*
 
+proc ::tincr::print_quiet { message quiet {newline ""} } {
```
This is not a good way to handle verbose logging IMO. Tincr already has a global boolean (tincr::debug) that allows the user to turn on debug logging. Perhaps you could use tincr::debug or add another global boolean called tincr::verbose to turn on the verbose logging you added. Then, I would change this method to look something like this:

```
proc ::tincr::print_verbose { message {newline 0} } {
    if {$::tincr::verbose} {
        puts -nonewline $message
        if {$newline} {
            puts {}
        }
    }
}
```

bradselw commented on tincr/io/design/tincr_checkpoints.tcl:
```diff
@@ -125,84 +125,39 @@ proc ::tincr::read_tcp {args} {
-		puts "Design linked successfully. ($link_runtime seconds)"
-	}
+	::tincr::print_quiet "Reading netlist..." $quiet
+	set edif_runtime [::tincr::time_command "read_edif $q ${filename}/netlist.edf"]
```
Use `report_runtime` instead of `time_command`

bradselw commented on tincr/io/design/tincr_checkpoints.tcl:
```diff
@@ -125,84 +125,39 @@ proc ::tincr::read_tcp {args} {
+	::tincr::print_quiet "Netlist added successfully. ($edif_runtime seconds)" $quiet
+			
+	::tincr::print_quiet "Linking design..." $quiet
+	set link_runtime [::tincr::time_command "link_design $q -part $part"]
```
Use `report_runtime` instead of `time_command`

bradselw commented on tincr/io/design/tincr_checkpoints.tcl:
```diff
@@ -125,84 +125,39 @@ proc ::tincr::read_tcp {args} {
 	# set_property IS_ENABLED 0 $placer_checks
+		
+	::tincr::print_quiet "Reading design constraints..." $quiet
+	set constraint_runtime [::tincr::time_command "read_xdc $q -no_add ${filename}/constraints.xdc"]
```
Use `report_runtime` instead of `time_command`

bradselw commented on tincr/io/design/tincr_checkpoints.tcl:
```diff
@@ -125,84 +125,39 @@ proc ::tincr::read_tcp {args} {
-		puts "Design constraints added successfully. ($constraint_runtime seconds)"
-	}
+	::tincr::print_quiet "Reading placement constraints..." $quiet
+	set place_runtime [::tincr::time_command "read_xdc $q -no_add ${filename}/placement.xdc"]
```
Use `report_runtime` instead of `time_command`

bradselw commented on tincr/io/design/tincr_checkpoints.tcl:
```diff
@@ -125,84 +125,39 @@ proc ::tincr::read_tcp {args} {
-		puts "Placement constraints added successfully... ($place_runtime seconds)"
-	}
+	::tincr::print_quiet "Reading routing constraints..." $quiet
+	set route_runtime [::tincr::time_command "read_xdc $q -no_add ${filename}/routing.xdc"]
```
Use `report_runtime` instead of `time_command`

ttown523 committed:
bug fixes and tincr import updates

ttown523 committed:
changed tabs to spaces in genCellLibrary.tcl script

ttown523 committed:
minor updates to rs routing file

ttown523 committed:
fixed a small bug

nelsobe commented:
Merging this in.  It looks good to me and Thomas.  I need it to continue work on my local changes.  Brad, please review if you have time.

bradselw commented:
Looks good to me :+1:
